src/pages/index.js

```diff
@@ -29,54 +32,153 @@ const enableValidation = (config) => {
+      },
+    },
+    cardListElement
+  );
+  cardSection.renderItems();
+});
+
+const createCard = (cardItem) => {
+  const card = new Card(
+    cardItem,
+    "#card-template",
+    () => {
+      imgPopup.open(cardItem);
+    },
+    () => {
+      deleteConfirmPopup.open(card);
+    }
+  );
   return card.getView();
 };
 
+//VAR
+
+let cardSection;
+let initialUserData;
+let currentAvatar;
+let profileData;
+let newCardData;
+
+//Popups
+
 const imgPopup = new PopupWithImages("#card-picture-modal");
 imgPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    data: initialCards,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      cardSection.addItem(cardElement);
-    },
-  },
-  cardListElement
+const profilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileUpdate
 );
-cardSection.renderItems();
-
-const userInfo = new UserInfo(".profile__name", ".profile__job");
-
-const profilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
-  profilePopup.close();
-  userInfo.setUserInfo(data);
-});
 profilePopup.setEventListeners();
 
-const newCardPopup = new PopupWithForm("#card-add-modal", (data) => {
-  console.log(data);
-  const cardFormElement = createCard(data);
-  cardSection.addItem(cardFormElement);
-  newCardPopup.close();
-});
+const newCardPopup = new PopupWithForm("#card-add-modal", handleAddCard);
 newCardPopup.setEventListeners();
 
+export const deleteConfirmPopup = new PopupConfirmation(
+  "#modal-delete",
+  handleDeleteConfirmation
+);
+deleteConfirmPopup.setEventListeners();
+
+const avatarPopup = new PopupWithForm("#modal-avatar", handleUpdateAvatar);
+avatarPopup.setEventListeners();
+
+//HANDLES
+
+function handleDeleteConfirmation(card, id) {
+  card.handleDeleteCard(id);
+}
+
+function handleProfileUpdate(userData) {
+  initialUserData.setUserInfo(userData);
+  profilePopup.setLoading(true);
+  api
+    .updateUserData(userData)
+    .then(() => profilePopup.close())
+    .catch((err) => console.error(`${err}, Failed to update Avatar`))
+    .finally(() => profilePopup.setLoading(false));
+}
+
+function handleAddCard(cardData) {
+  newCardPopup.setLoading(true);
+  api
+    .createCard(cardData)
+    .then((result) => {
+      if (result.ok) return result.json();
+    })
+    .then((data) => {
+      // console.log(`**handleAddCard**${data}`);
+      const cardFormElement = createCard(data);
+      cardSection.addItem(cardFormElement);
+
+      // ************************************************************************** need to pull id from server now or before like and delete
+
+      newCardPopup.close();
+    })
+    .catch((err) => console.error(`${err}, Failed to update Avatar`))
+    .finally(() => newCardPopup.setLoading(false));
+}
+
+function handleUpdateAvatar(input) {
+  avatarPopup.setLoading(true);
+  api
+    .updateAvatar(input.link)
+    .then(() => {
+      avatarProfileImage.src = input.link;
```

Comment:
Class `UserInfo` should set the entire user’s data (including the **avatar**)

Reply:
Updates in UserInfo now